src/main/java/org/sagebionetworks/web/client/widget/table/v2/results/QueryResultEditorWidget.java

```diff
@@ -367,11 +369,12 @@ private void doHideEditor() {
 
   public static boolean isTableTypeQueryResultEditable(TableType tableType) {
     // Datasets should not be editable (SWC-5870, SWC-5903)
-    return !(
-      tableType.equals(TableType.dataset) ||
-      tableType.equals(TableType.dataset_collection) ||
-      tableType.equals(TableType.materialized_view)
-    );
+    return isTableTypeQueryResultEditable(tableType.getClazz());
+  }
+
+  public static boolean isTableTypeQueryResultEditable(Class<?> clazz) {
+    // EntityRefCollectionView results are not editable, even if the user has permissions, because rows may reference immutable versions (SWC-5870, SWC-5903)
+    return clazz.equals(TableEntity.class) || clazz.equals(EntityView.class);
```

Comment:
Easier to list those table types that can be edited...notably, the result changes for `SubmissionView` -- I don't think this was functional in that case.